jena-fuseki2/jena-fuseki-mod-geosparql/src/test/java/org/apache/jena/fuseki/mod/geosparql/TestFMod_SpatialIndexer.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.jena.geosparql.spatial.index.v2.SpatialIndexUtils;
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.DatasetGraph;
+import org.apache.jena.sparql.core.DatasetGraphFactory;
+import org.apache.jena.sparql.core.Quad;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.locationtech.jts.geom.Envelope;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class TestFMod_SpatialIndexer {
+    private WebDriver driver;
+    private JavascriptExecutor js;
+
+    private static Envelope queryEnvelope = new Envelope(-180, 180, -90, 90);
+    private DatasetGraph dsg;
+    private SpatialIndex spatialIndex;
+    private Node graphName1 = NodeFactory.createURI("http://www.example.org/graph1");
+
+    @Before
+    public void setUp() throws IOException, SpatialIndexException {
+        dsg = DatasetGraphFactory.create();
+        setupTestData(dsg);
+
+        spatialIndex = SpatialIndexUtils.buildSpatialIndex(dsg);
+
+        String[] argv = new String[] { "--empty" };
+        FusekiServer server = FusekiMain.builder(argv)
+            .add("test", dsg)
+            .build();
+        server.start();
+        int port = server.getPort();
+        String serverURL = "http://localhost:" + port + "/";
+        String siteUrl = serverURL + "test/spatial";
+
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--headless=new"); // use "new" headless mode (better support)
+        options.addArguments("--no-sandbox");
+        options.addArguments("--disable-dev-shm-usage");
+
+        WebDriverManager.chromedriver().setup(); // Automatically downloads and sets path
+        driver = new ChromeDriver(options);
+        driver.get(siteUrl);
+        js = (JavascriptExecutor) driver;
+    }
+
+    private void setupTestData(DatasetGraph dsg) {
+        // Fill the graph with a few geometries; spatial index construction will derive the SRS from them.
+        Envelope envelope = new Envelope(-175, 175, -85, 85);
+
+        Map<GeometryType, Number> conf = new HashMap<>();
+        conf.put(GeometryType.POINT, 1);
+
+        // Generate geometries into the default graph and a named graph
+        GeometryGenerator.generateGraph(dsg.getDefaultGraph(), envelope, conf);
+
+        conf.put(GeometryType.POLYGON, 1);
+        GeometryGenerator.generateGraph(dsg.getGraph(graphName1), envelope, conf);
+    }
+
+    @Test
+    @Ignore // Ignored for new because this test requires a local Chrome browser.
```

Comment:
Remove @Ignore to run this test; but requires a local chrome. This test could be improved e.g. by starting chrome in docker via testcontainers. Perhaps it should be migrated to the cypress setup (if possible).